### ISSUE TITLE
Add logging for controller creation

### ISSUE THREAD

**Issue by WBarge** [aspnet/AspNetCore#11005]
I am seeing a slow down in the processing of a request before it is handed over to the controller.
if you look in the log you will see 1.6sec time elapse between "Execution plan of "result" filters" and "Attempting to bind parameter" (timestamp 2019-06-07 14:45:36.810 -07:00 in the log below)

My logic is taking 58.2462ms. per the logs which is ok but over all the request is taking 1759.5546ms

here is the log of the request
```
14:45:36.767 -07:00 [Information] Request starting HTTP/1.1 GET http://localhost:44385/api/Client/GetClientsForTimeKeeper?timeKeeperId=-1&searchStr=morgan&maxNumOfRecordsReturned=50  
===> 2019-06-07 14:45:36.768 -07:00 [Debug] Connection id ""0HLNBI59LIRA8"" received FIN.
===> 2019-06-07 14:45:36.773 -07:00 [Debug] Connection id ""0HLNBI59LIRA8"" disconnecting.
===> 2019-06-07 14:45:36.799 -07:00 [Debug] Request successfully matched the route with name 'null' and template '"api/Client/GetClientsForTimeKeeper"'
===> 2019-06-07 14:45:36.800 -07:00 [Debug] Connection id ""0HLNBI59LIRA8"" sending FIN because: ""The Socket transport's send loop completed gracefully.""
===> 2019-06-07 14:45:36.802 -07:00 [Information] Route matched with "{action = \"GetClientsForTimeKeeper\", controller = \"Client\"}". Executing controller action with signature "System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] GetClientsForTimeKeeper(System.String, System.String, Int32)" on controller "Minisoft.Api.Controllers.API.ClientController" ("Minisoft.Api").
===> 2019-06-07 14:45:36.804 -07:00 [Debug] Connection id ""0HLNBI59LIRA8"" stopped.
===> 2019-06-07 14:45:36.805 -07:00 [Debug] Execution plan of "authorization" filters (in the following order): ["None"]
===> 2019-06-07 14:45:36.806 -07:00 [Debug] Execution plan of "resource" filters (in the following order): ["None"]
===> 2019-06-07 14:45:36.807 -07:00 [Debug] Execution plan of "action" filters (in the following order): ["None"]
===> 2019-06-07 14:45:36.809 -07:00 [Debug] Execution plan of "exception" filters (in the following order): ["None"]
===> 2019-06-07 14:45:36.810 -07:00 [Debug] Execution plan of "result" filters (in the following order): ["None"]
===> 2019-06-07 14:45:38.442 -07:00 [Debug] Attempting to bind parameter '"timeKeeperId"' of type '"System.String"' ...
===> 2019-06-07 14:45:38.444 -07:00 [Debug] Attempting to bind model of type '"System.String"' using the name '"timeKeeperId"' in request data ...
===> 2019-06-07 14:45:38.445 -07:00 [Debug] Done attempting to bind model of type '"System.String"' using the name '"timeKeeperId"'.
===> 2019-06-07 14:45:38.446 -07:00 [Debug] Done attempting to bind parameter '"timeKeeperId"' of type '"System.String"'.
===> 2019-06-07 14:45:38.447 -07:00 [Debug] Attempting to bind parameter '"searchStr"' of type '"System.String"' ...
===> 2019-06-07 14:45:38.448 -07:00 [Debug] Attempting to bind model of type '"System.String"' using the name '"searchStr"' in request data ...
===> 2019-06-07 14:45:38.449 -07:00 [Debug] Done attempting to bind model of type '"System.String"' using the name '"searchStr"'.
===> 2019-06-07 14:45:38.450 -07:00 [Debug] Done attempting to bind parameter '"searchStr"' of type '"System.String"'.
===> 2019-06-07 14:45:38.452 -07:00 [Debug] Attempting to bind parameter '"maxNumOfRecordsReturned"' of type '"System.Int32"' ...
===> 2019-06-07 14:45:38.453 -07:00 [Debug] Attempting to bind model of type '"System.Int32"' using the name '"maxNumOfRecordsReturned"' in request data ...
===> 2019-06-07 14:45:38.454 -07:00 [Debug] Done attempting to bind model of type '"System.Int32"' using the name '"maxNumOfRecordsReturned"'.
===> 2019-06-07 14:45:38.456 -07:00 [Debug] Done attempting to bind parameter '"maxNumOfRecordsReturned"' of type '"System.Int32"'.
===> 2019-06-07 14:45:38.457 -07:00 [Information] Executing action method "Minisoft.Api.Controllers.API.ClientController.GetClientsForTimeKeeper (Minisoft.Api)" - Validation state: Valid
===> 2019-06-07 14:45:38.515 -07:00 [Debug] Time To get data from the Manager 58
===> 2019-06-07 14:45:38.515 -07:00 [Debug] Time To transform the data 0
===> 2019-06-07 14:45:38.515 -07:00 [Information] Returning values 'System.Collections.Generic.List`1[Minisoft.Api.Models.Responses.ClientResponse]' 
===> 2019-06-07 14:45:38.516 -07:00 [Information] Executed action method "Minisoft.Api.Controllers.API.ClientController.GetClientsForTimeKeeper (Minisoft.Api)", returned result "Microsoft.AspNetCore.Mvc.OkObjectResult" in 58.2462ms.
===> 2019-06-07 14:45:38.517 -07:00 [Debug] List of registered output formatters, in the following order: ["Microsoft.AspNetCore.Mvc.Formatters.HttpNoContentOutputFormatter", "Microsoft.AspNetCore.Mvc.Formatters.StringOutputFormatter", "Microsoft.AspNetCore.Mvc.Formatters.StreamOutputFormatter", "Microsoft.AspNetCore.Mvc.Formatters.JsonOutputFormatter"]
===> 2019-06-07 14:45:38.518 -07:00 [Debug] No information found on request to perform content negotiation.
===> 2019-06-07 14:45:38.519 -07:00 [Debug] Attempting to select an output formatter without using a content type as no explicit content types were specified for the response.
===> 2019-06-07 14:45:38.521 -07:00 [Debug] Attempting to select the first formatter in the output formatters list which can write the result.
===> 2019-06-07 14:45:38.522 -07:00 [Debug] Selected output formatter '"Microsoft.AspNetCore.Mvc.Formatters.JsonOutputFormatter"' and content type '"application/json"' to write the response.
===> 2019-06-07 14:45:38.523 -07:00 [Information] Executing ObjectResult, writing value of type '"System.Collections.Generic.List`1[[Minisoft.Api.Models.Responses.ClientResponse, Minisoft.Api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]"'.
===> 2019-06-07 14:45:38.524 -07:00 [Information] Executed action "Minisoft.Api.Controllers.API.ClientController.GetClientsForTimeKeeper (Minisoft.Api)" in 1712.699ms
===> 2019-06-07 14:45:38.525 -07:00 [Debug] Connection id ""0HLNBI59LIRA9"" completed keep alive response.
===> 2019-06-07 14:45:38.526 -07:00 [Information] Request finished in 1759.5546ms 200 application/json; charset=utf-8
```
### Expected behavior
The transition between the filters and the binding of the parameters should be in milliseconds not seconds

**Comment by rynowak** [comment]
The obvious thing to suspect here is that there's some expensive work happening inside the controller creation part of the pipeline. Does your controller depend on things that are expensive to create?

This is the code that does the logging here:  https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs#L161
Then the controller is created: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs#L67
Then arguments are bound: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs#L71

**Comment by rynowak** [comment]
It looks like it would be beneficial for us to add logging around controller creation.

**Comment by WBarge** [comment]
adding logging to the resolution of objecting being injected into a controller would have been very helpful here.  Thank you for your comments.  Time is now in an acceptable range

**Comment by rynowak** [comment]
Thanks @WBarge - I'm going to reopen this to track us adding logging to the controller factory.

**Comment by alefranz** [comment]
Would it be ok if I send a PR for this? or are you working on it @ryanbrandenburg ?

I had a look and it feels like the best place to log it is in `ControllerActionInvoker` because, even if it doesn't allow to get more fine grained info, it would work for any controller factory and would not required a closure on the factory delegate.

**Comment by rynowak** [comment]
Sure, @alefranz that would be great. If you end up doing this in `ControllerActionInvoker` can you also do the same of `PageActionInvoker`?

**Comment by alefranz** [comment]
ah yes! good point.
This is just for logging and does not include changes to the diagnostic listener, right?

**Comment by rynowak** [comment]
Correct

**Comment by alefranz** [comment]
I still have a few questions but I put them on the PR #11526. Thank you.

**Comment by alefranz** [comment]
I think this is fixed by #11526 .
Probably I got something wrong in the PR description and it didn't get automatically closed.

**Comment by rynowak** [comment]
Yup! Thanks again!
